backend/Controllers/UsersController.cs

```diff
@@ -65,26 +69,38 @@ public async Task<IActionResult> CreateAsync([FromForm] CreateUserDTO userDTO, C
         {
             //Def not the best way to handle it, but can't extract content-type header as we're getting the form data
             //and I'm not sure if using FileExtensionContentTypeProvider is beneficial in this case
-            if(userDTO.AvatarFile.ContentType != "image/bmp" || 
+            if(userDTO.AvatarFile.ContentType != "image/bmp" ||
             userDTO.AvatarFile.ContentType != "image/jpeg" ||
             userDTO.AvatarFile.ContentType != "image/png")
             {
+                _logger.LogError("Incorrect image format");
                 return BadRequest("Invalid image format, only bmp, jpg and png are supported.");
             }
             avatarId = await AddAvatarAsync(userDTO.AvatarFile, cancellationToken);
+
+            if (avatarId == null)
+            {
+
+                _logger.LogError("Error adding an avatar.");
+                return StatusCode(StatusCodes.Status500InternalServerError, "Server error");
+            }
         }
         var contact = _mapper.Map<CreateUserDTO, User>(userDTO);
         contact.AvatarId = avatarId;
 
         var createdUser = await _appDbContext.Users.AddAsync(contact);
         await _appDbContext.SaveChangesAsync(cancellationToken);
-        
+
+        _logger.LogInformation("Changes saved successfully");
+
         return Ok(createdUser.Entity.Id);
     }
 
     private async Task<int> AddAvatarAsync(IFormFile avatarFile, CancellationToken cancellationToken)
     {
         Guid avatarGuid = new Guid();
+
+        _logger.LogInformation($"Created a new GUID for the avatar: {avatarGuid}");
```

Comment:
Redundant log, I suggest adding information about entity Id in the "Changes saved successfully!" log message.